runner/android_test_orchestrator/java/androidx/test/orchestrator/AndroidTestOrchestrator.java

```diff
@@ -401,7 +400,27 @@ private String getOutputFile() {
     if (null == test) {
       return TEST_COLLECTION_FILENAME;
     } else {
-      return String.format(TEST_RUN_FILENAME, test);
+      String filename = test; // Copy it so we don't modify the original.
```

Comment:
String are immutable in Java, one cannot "modify the original".

Reply:
@ralf-at-android it's a class variable though. So I think the idea is that Bartek wants to do operations without having to worry about accidentally modifying the class variable.

Reply:
Probably simplest is to take a step back and remove the mutation altogether. Then the filename local can be created few lines later, and there's no comment.